### PR TITLE
tests: Implemented more tests for fs-v1*.go

### PR DESCRIPTION
**Pull request by vadmeste**
- Wrote more tests for all private functions.

### CODE REVIEW

**Comment by harshavardhana** [comment]
Move these tests to distributed branch @vadmeste we can avoid rebasing.

File: cmd/fs-v1-metadata_test.go [hunk 1]
```diff
@@ -64,3 +66,91 @@ func TestHasExtendedHeader(t *testing.T) {
 		}
 	}
 }
+
+// ReadFsMetadata
```
Comment (harshavardhana) [inline cmd/fs-v1-metadata_test.go]: A better comment would make more sense here..

File: cmd/fs-v1-metadata_test.go [hunk 2]
```diff
@@ -64,3 +66,91 @@ func TestHasExtendedHeader(t *testing.T) {
...
+	fsStorage := fs.storage.(*posix)
+	naughty := newNaughtyDisk(fsStorage, nil, errFaultyDisk)
+	fs.storage = naughty
+	if _, err = readFSMetadata(fs.storage, ".minio.sys", fsPath); errorCause(err) != errFaultyDisk {
+		t.Fatal("Should fail", err)
+	}
+
+}
+
+// WriteFsMetadata
```
Comment (harshavardhana) [inline cmd/fs-v1-metadata_test.go]: same as above.

File: cmd/fs-v1-multipart-common_test.go [hunk 3]
```diff
@@ -0,0 +1,165 @@
...
+
+	// isBucketExist with a faulty disk
+	fsStorage := fs.storage.(*posix)
+	naughty := newNaughtyDisk(fsStorage, nil, errFaultyDisk)
+	fs.storage = naughty
+	if found := fs.isBucketExist(bucketName); found {
+		t.Fatal("isBucketExist should return false because it is wired to a corrupted disk")
+	}
+
+	removeAll(disk)
```
Comment (harshavardhana) [inline cmd/fs-v1-multipart-common_test.go]: defer this earlier. ?

**Comment by vadmeste** [comment]
@harshavardhana, could you wait a little until I put all the comments and push again ? also there are some additional tests but they fail with faulty disks probably because of some bugs in the code

File: cmd/object-api-listobjects_test.go [hunk 4]
```diff
@@ -437,7 +437,7 @@ func testListObjects(obj ObjectLayer, instanceType string, t TestErrHandler) {
 		{"test-bucket-list-object", "asia", "europe-object", "", 0, ListObjectsInfo{}, fmt.Errorf("Invalid combination of marker '%s' and prefix '%s'", "europe-object", "asia"), false},
 		// Setting a non-existing directory to be prefix (12-13).
 		{"empty-bucket", "europe/france/", "", "", 1, ListObjectsInfo{}, nil, true},
-		{"empty-bucket", "europe/tunisia/", "", "", 1, ListObjectsInfo{}, nil, true},
+		{"empty-bucket", "africa/tunisia/", "", "", 1, ListObjectsInfo{}, nil, true},
```
Comment (harshavardhana) [inline cmd/object-api-listobjects_test.go:440]: I see :+1:

**Review by harshavardhana — APPROVED** [review]

**Pull request merged**